src/contract.ts

```diff
@@ -73,7 +202,7 @@ export class Contract {
    * @param method  - name of the method
    * @param args - arguments that are passed to the method
    */
-  private validateMethodAndArgs(type: 'INVOKE' | 'CALL', method: string, args: Args = {}) {
+  protected validateMethodAndArgs(type: 'INVOKE' | 'CALL', method: string, args: Args = {}) {
```

Comment:
can we please add a new validation that the provided object doesn't contain more keys than needed?
eg:
```
// ABI expects:
{
  account: address
}
// provided: 
{
  user: "0x5"
}
```
should throw as "user" is not expected in ABI. This should throw. Current behavior is that it just returns with `["0x0"]` as account wasn't provided, so it assumes 0 (which is fine, but confusing without the error I'm talking about).
Would solve #140

Reply:
This will change with the new method signature of the Contract since the arguments are going to be positional.